magefile.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2023 uwu-tools Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build mage
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/magefile/mage/sh"
+	"sigs.k8s.io/release-utils/mage"
+
+	"github.com/uwu-tools/magex/pkg"
+)
+
+// Default target to run when none is specified
+// If not set, running mage will list available targets
+var Default = Verify
+
+const (
+	binDir     = "bin"
+	moduleName = "github.com/uwu-tools/gh-jira-issue-sync"
+	scriptDir  = "scripts"
+
+	// Versions.
+	golangciVersion = "v1.50.1"
+	minGoVersion    = "1.20"
+
+	// Environment variables.
+	envLDFLAGS      = "GHJIRA_LDFLAGS"
+	envMinGoVersion = "MIN_GO_VERSION"
+
+	// Test variables.
+	coverMode            = "atomic"
+	coverProfileFilename = "unit-coverage.out"
+)
+
+// All runs all targets for this repository
+func All() error {
+	if err := Verify(); err != nil {
+		return err
+	}
+
+	if err := Test(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Test runs various test functions
+func Test() error {
+	// TODO(mage): Upstream should support setting covermode/coverprofile.
+	return sh.RunV(
+		"go",
+		"test",
+		"-v",
+		"-covermode", coverMode,
+		"-coverprofile", coverProfileFilename,
+		"./...",
+	)
+}
+
+// Verify runs repository verification scripts
+func Verify() error {
+	fmt.Println("Ensuring mage is available...")
+	if err := pkg.EnsureMage(""); err != nil {
+		return err
+	}
+
+	// TODO(mage): Support verifying headers.
+	/*
+		fmt.Println("Running copyright header checks...")
+		if err := mage.VerifyBoilerplate("v0.2.5", binDir, boilerplateDir, false); err != nil {
+			return err
+		}
+	*/
+
+	// TODO(mage): Support verifying external dependencies.
+	/*
+		fmt.Println("Running external dependency checks...")
+		if err := mage.VerifyDeps("v0.3.0", "", "", true); err != nil {
+			return err
+		}
+	*/
+
+	// TODO(mage): Support verifying go module.
+	// TODO(mage): Upstream should support setting minimum version.
+	/*
+		os.Setenv(envMinGoVersion, minGoVersion)
+		fmt.Println("Running go module linter...")
+		if err := mage.VerifyGoMod(scriptDir); err != nil {
+			return err
+		}
+	*/
+
+	fmt.Println("Running golangci-lint...")
+	if err := mage.RunGolangCILint(golangciVersion, false); err != nil {
+		return err
+	}
+
+	if err := Build(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Build runs go build
+func Build() error {
```

Comment:
Build doesn't work on OSX, created an issue for it: https://github.com/uwu-tools/gh-jira-issue-sync/issues/166